pcsx2/gui/Panels/GSWindowPanel.cpp

```diff
@@ -136,46 +189,63 @@ void Panels::GSWindowSettingsPanel::AppStatusEvent_OnSettingsApplied()
+		m_combo_ScalingType->Enable(conf.AspectRatio != AspectRatio_Stretch);
+		m_combo_AspectRatio->SetSelection(enum_cast(conf.AspectRatio));
+		m_combo_FMVAspectRatioSwitch->SetSelection(enum_cast(conf.FMVAspectRatioSwitch));
+		m_combo_FMVAspectRatioSwitch->Enable(conf.AspectRatio != AspectRatio_4_3);
+		m_text_WindowWidth->ChangeValue(wxsFormat(L"%d", conf.WindowSize.GetWidth()));
+		m_text_WindowHeight->ChangeValue(wxsFormat(L"%d", conf.WindowSize.GetHeight()));
+		m_text_Zoom->ChangeValue(conf.Zoom.ToString());
+
+		m_check_SizeLock->SetValue(conf.DisableResizeBorders);
+		m_check_HideMouse->SetValue(conf.AlwaysHideMouse);
+		m_check_CloseGS->SetValue(conf.CloseOnEsc);
+		m_check_ScalingCompensation->SetValue(conf.DisableScalingCompensation);
+		m_check_ScalingCompensation->Enable((conf.AspectRatio != AspectRatio_Stretch &&
+											conf.ScalingType != ScalingType_Fit) ||
+											conf.AspectRatio == AspectRatio_Frame);
+
+		m_check_Fullscreen->SetValue(conf.DefaultToFullscreen);
+		m_check_DclickFullscreen->SetValue(conf.IsToggleFullscreenOnDoubleClick);
 	}
 
 	m_combo_vsync->SetSelection(enum_cast(configToApply.EmuOptions.GS.VsyncEnable));
-	m_combo_vsync->Enable( !configToApply.EnablePresets );//grayed-out when presets are enabled
+	m_combo_vsync->Enable(!configToApply.EnablePresets); // grayed-out when presets are enabled
+}
+
+void Panels::GSWindowSettingsPanel::ScalingTypeChanged(wxCommandEvent &event)
+{
+	m_combo_ScalingType->Enable(m_combo_AspectRatio->GetSelection() != 0);
+	m_check_ScalingCompensation->Enable((m_combo_AspectRatio->GetSelection() != 0 &&
+		m_combo_ScalingType->GetSelection() != 0) || m_combo_AspectRatio->GetSelection() == 3);
+	m_combo_FMVAspectRatioSwitch->Enable(m_combo_AspectRatio->GetSelection() != 1);
+
+	event.Skip();
 }
 
 void Panels::GSWindowSettingsPanel::Apply()
 {
 	AppConfig::GSWindowOptions& appconf( g_Conf->GSWindow );
 	Pcsx2Config::GSOptions& gsconf( g_Conf->EmuOptions.GS );
 
-	appconf.CloseOnEsc				= m_check_CloseGS	->GetValue();
-	appconf.DefaultToFullscreen		= m_check_Fullscreen->GetValue();
-	appconf.AlwaysHideMouse			= m_check_HideMouse	->GetValue();
-	appconf.DisableResizeBorders	= m_check_SizeLock	->GetValue();
+	appconf.ScalingType = (ScalingTypes)m_combo_ScalingType->GetSelection();
+	appconf.AspectRatio = (AspectRatioType)m_combo_AspectRatio->GetSelection();
+	appconf.FMVAspectRatioSwitch = (FMVAspectRatioSwitchType)m_combo_FMVAspectRatioSwitch->GetSelection();
+	appconf.Zoom = Fixed100::FromString(m_text_Zoom->GetValue());
 
-	appconf.AspectRatio		= (AspectRatioType)m_combo_AspectRatio->GetSelection();
-	appconf.Zoom			= Fixed100::FromString( m_text_Zoom->GetValue() );
-
-	gsconf.VsyncEnable = static_cast<VsyncMode>(m_combo_vsync->GetSelection());
+	appconf.DisableResizeBorders = m_check_SizeLock	->GetValue();
+	appconf.AlwaysHideMouse = m_check_HideMouse	->GetValue();
+	appconf.CloseOnEsc = m_check_CloseGS	->GetValue();	
```

Comment:
Trailing whitespace. Also, you should probably remove the whitespace before `->` on these lines.